### PR TITLE
Import style editor actor via prosthesis.

### PR DESCRIPTION
**Pull request by ochameau**
Here is a tentative to load styleeditor actor via prosthesis.
Unfortunately, for some reason it doesn't work. We end up with a style editor panel with infinite loading throbbers.

### CODE REVIEW

File: prosthesis/content/dbg-styleeditor-actors.js [hunk 1]
```diff
@@ -0,0 +1,741 @@
...
+  },
+
+  /**
+   * Clear all the current stylesheet actors in map.
+   */
+  _clearStyleSheetActors: function() {
+    for (let actor in this._sheets) {
+      this.releaseActor(this._sheets[actor]);
+    }
+    this._sheets.clear();
```
Comment (mykmelez) [inline prosthesis/content/dbg-styleeditor-actors.js:262]: The problem turns out to be simple: according to the [Map documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map), Map.clear() was implemented in Mozilla 19, so it isn't available on the b2g18 branch, where we're using this code, which causes this call to throw an exception. The fix is to catch the exception and use a less efficient but backward-compatible loop to clear the sheets (both here and in _disconnect_):

```
    try {
      this._sheets.clear();
    } catch(e) {
      // Mozilla 18 doesn't have Map.clear(), which was introduced in 19.
      for (let actor in this._sheets) {
        this._sheets.delete(actor);
      }
    }
```

**Comment by mykmelez** [comment]
The only other issue with this change is that the line-endings in dbg-styleeditor-actors.js are Windows-style \r\n pairs instead of conventional Unix-style \n characters. I'll merge the change with fixes for both issues.

**Comment by ochameau** [comment]
Oh, awesome! Thanks for having looked at that, I wonder how I missed that, was it silently failing ?

@harthur @campd Look, we have the style editor for gecko18 \o/
https://github.com/mozilla/r2d2b2g/blob/master/prosthesis/content/dbg-styleeditor-actors.js

**Comment by mykmelez** [comment]
> Oh, awesome! Thanks for having looked at that, I wonder how I missed that, was it silently failing ?

Nope, it was throwing an exception to the web consoles I connected to the main process and shell.xul! But perhaps that exception wasn't showing up in the Error Console?

**Pull request merged**